docs/generics/index.md

```diff
@@ -82,6 +83,31 @@ class FooString
   fun ref set(c: String val) => _c = c
 ```
 
+### Type parameter defaults
+
+Sometimes the same parameter type is used over and over again, and it is tedious to always specify it when using the generic class.
+The class `Bar` expects its type parameter to be a `val USize` by default:
+
+```pony
+class Bar[A: Any val = USize]
```

Comment:
Although it is implied by the default refcap of a `USize` will you change this to match how you wrote it above as `USize val`?

Reply:
I wasn't aware that the RHS is subject to "reference capability defaulting" again. So `[A: Any val = USize val]`?

Reply:
I don't think the RHS is subject to defaulting like my previous comment suggested. After looking at generic parameters in the stdlib and reconsidering this minor change I have reversed my opinion to this change NOT being needed.

The reason being that this style of the default refcap being repeated so `Any **val** = USize **val**` is not the norm in the stdlib. As well, I think the repetition, although superficially mirroring the above change, is distracting since `Any **val**` is already specified.

Reply:
This leads to a interesting question:

Is this legal, the default having a different ref cap? (I changed it so the default is valid, "val" is acceptable where a "box" is expected)?

```
class Bar[A: Any box = USize val]
```

If yes, it should probably be mentioned, including that `[A: Any box = USize]` means `[A: Any box = USize box]` and that it is usual to leave the second ref cap out.
If not, I would just keep this part as it is.

Reply:
You should run some tests because I do not know without looking if this is legal and works the way you mention. If it does work that way, I agree that adding additional information here would be useful.

Reply:
`class Bar[A: Any box = USize val]` is legal. I'll make a proposal in a separate commit.

Reply:
> including that [A: Any box = USize] means [A: Any box = USize box]

I don't think that's correct, it will still mean `USize val`. See this example on how `[A: Any tag = USize]` will fail to compile:

https://playground.ponylang.io/?gist=576c7950021c592647febc8016120e03

Reply:
Thank you @ergl. I've updated accordingly, hope I got it right this time.

Reply:
I think it looks fine now 👍